resources/qml/PrinterSelector/MachineSelector.qml

```diff
@@ -26,7 +26,14 @@ Cura.ExpandablePopup
 
     headerItem: Cura.IconWithText
     {
-        text: isNetworkPrinter ? Cura.MachineManager.activeMachineNetworkGroupName : Cura.MachineManager.activeMachineName
+        text:
+        {
+            if (isNetworkPrinter && Cura.MachineManager.activeMachineNetworkGroupName != "")
+            {
+                Cura.MachineManager.activeMachineNetworkGroupName
```

Comment:
You missed a return here. Fixed that for you in b1b61fa466eafa51121ce7a276dc2046d5f38ea7